src/overlay/TxAdverts.h

```diff
@@ -37,14 +36,14 @@ class TxAdverts
     RandomEvictionCache<Hash, uint32_t> mAdvertHistory;
     TxAdvertVector mOutgoingTxHashes;
     VirtualTimer mAdvertTimer;
-    std::weak_ptr<Peer> mWeakPeer;
+    std::function<void(std::shared_ptr<StellarMessage const>)> mSendCb;
 
     void rememberHash(Hash const& hash, uint32_t ledgerSeq);
     void flushAdvert();
     void startAdvertTimer();
 
   public:
-    TxAdverts(Application& app, std::weak_ptr<Peer> peer);
+    TxAdverts(Application& app);
```

Comment:
Should this class take an `OverlayAppConnector` instead of an `App`? It looks like `OverlayAppConnector` provides everything this class needs.

Reply:
TxAdverts needs `Application` to construct timers, so I kept references to mApp here. I think this is okay though, since everything inside TxAdverts must be run from the main thread (even after the incoming background changes).